BUILD.bazel

```diff
@@ -0,0 +1,13 @@
+licenses(["notice"])  # 3-Clause BSD
+
+exports_files(["LICENSE.MIT"])
+
+cc_library(
+    name = "json",
+    hdrs = glob([
+        "include/**/*.hpp",
```

Comment:
From the [docs](https://docs.bazel.build/versions/master/be/functions.html#glob): "This pattern should be avoided if possible, as it can reduce build incrementality and therefore increase build times." A better approach would be to generate the list of sources and headers from CMake (that way, you don't need to repeat all of it in Bazel), i.e: https://github.com/googleapis/google-cloud-cpp/blob/8e8dd3d83ec27b7a5176f9fe43cdd93aa5ace3dc/cmake/CreateBazelConfig.cmake#L43

Reply:
I changed this in https://github.com/nlohmann/json/pull/3709